File: informatykaEuropejczyka/Rozdzial2/prog2_1.cpp
//algorytm sprawdzajacy, czy punkt P lezy na prostej 3x+4y–4 = 0
#include <iostream>
#include <cmath>
using namespace std;

bool szukaj (double x0, double y0)
{
 double d=abs(3*x0+4*y0-4)/sqrt(pow(3.0,2)+pow(4.0,2));
 if (d==0) return true;
 else return false;
}

int main()
{
 double x0, y0;
 cout<<"podaj wspolrzedne punktu: ";
 cin>>x0>>y0;
 if (szukaj(x0,y0)) cout<<"punkt P=("<<x0<<";"<<y0<<") lezy na prostej 3x+4y-4=0"<<endl;
 else cout<<"punkt P=("<<x0<<";"<<y0<<") nie lezy na prostej 3x+4y-4=0"<<endl;
 return 0;
}

File: informatykaEuropejczyka/Rozdzial2/prog2_1.py
#algorytm sprawdzajacy, czy punkt P lezy na prostej 3x+4y-4 = 0

from math import *
       
def szukaj (x0, y0):
    d=fabs(3*x0+4*y0-4)/sqrt(3.0**2+4.0**2)
    if d==0:
        return True
    return False

#print(szukaj(2.5,3))

